x-pack/plugins/fleet/server/services/elastic_agent_manifest.ts

```diff
@@ -371,6 +395,12 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
+            - name: KUBERNETES_NODE_NAME
+              # Special environment indicator for Universal Profiling.
+              # If you are not using this integration, then this can be removed.
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
```

Comment:
Do we have a guidance on commented by default vs uncommented by default?
What I mean is: are special settings, needed only by certain integrations, going to be unset by default (we do this for securityContext below), or set?
If there's a preferred mode, we should be consistent in using it, otherwise this looks fine as the added environment variable makes no harm.

Note on `spec.nodeName`: it has been brought to my attention that not all Kubernetes distributions expose this field in the [Pod manifest](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) controlled by the DaemonSet.
I didn't think it was possible TBH, as it is a violation of the conformity, but it is what it is.

Reply:
on `spec.nodeName`: Looking up at the other environment definitions, like `NODE_NAME`,  they are also using `spec.nodeName`. For our `containermetadata` collector to work `KUBERNETES_NODE_NAME` needs to be set.

on guidance: I'm not aware of special guidance. But I tried to follow the style I did find this in this part of the code. Non-critical or non-security related updated can be added, while security related changes are encouraged to be explicitly commented to make the user aware.
Overall I would prefer if this `manifest.yml` is generated in a automated way depending on the selected integrations. But looking at the time concerns to get this into 8.9 before FF, I don't see this as an option or happening.

Reply:
FWIU on https://github.com/elastic/kibana/pull/159781#issuecomment-1593158830 we should comment by default all Universal Profiling options.
Users will have to de-comment when enabling the integration.